### PR TITLE
[modules] Fixing including JS modules that include arc modules

### PR DESCRIPTION
**Pull request by brianjjones**
Signed-off-by: Brian J Jones <brian.j.jones@intel.com>

### CODE REVIEW

**Comment by jprestwo** [comment]
+1 LGTM

**Comment by jimmy-huang** [comment]
I see a couple other places in the Makefile that uses $(JS) instead of js.tmp, shouldn't those be changed as well? maybe fix that in this patch too, and instead of hard-coding js.tmp, create another variable called JS_TMP or something

@./scripts/convert.sh $(JS)

**Comment by jprestwo** [comment]
@jimmy-huang, in some cases we do want to pass in just the $(JS) file. There are two "script" type parameters, JS_OUT is for the output JS file (after we append JS requires) and SCRIPT is for the original script.

BUT scripts/convert.sh should be using js.tmp otherwise you wont get any of the appended JS requires. @brianjjones said this patch fixed the issue but maybe it was not using convert.sh in his test?

**Comment by jimmy-huang** [comment]
yeah, i think convert.sh need to use js.tmp for Linux builds as well

**Comment by brianjjones** [comment]
This update should take care of all that.

File: Makefile [hunk 1]
```diff
@@ -353,17 +354,17 @@ ifeq ($(SNAPSHOT), on)
...
 	else \
-		cat js.tmp > outdir/jsgen.tmp; \
+		cat $(JS_TMP) > outdir/jsgen.tmp; \
 	fi
 	@outdir/snapshot/snapshot outdir/jsgen.tmp > outdir/include/zjs_snapshot_gen.h
 else
 	@echo Creating C string from JS application...
 ifeq ($(BOARD), linux)
-	@./scripts/convert.sh $(JS) outdir/include/zjs_script_gen.h
+	@./scripts/convert.sh $(JS_TMP) outdir/include/zjs_script_gen.h
```
Comment (jimmy-huang) [inline Makefile]: remove the linux then, since they are the same now

**Comment by jimmy-huang** [comment]
+1

**Pull request merged**